page_iterator.go

```diff
@@ -107,31 +105,32 @@ func (pI *PageIterator) SetReqOptions(reqOptions []abstractions.RequestOption) {
 	pI.reqOptions = reqOptions
 }
 
-func (pI *PageIterator) hasNext() bool {
-	if pI.currentPage == nil || pI.currentPage.getNextLink() == nil {
-		return false
+func (pI *PageIterator) next() error {
```

Comment:
sorry for making you remove hasnext, please add it back and make it public following our latest internal conversations.

Reply:
Resolving this for the same reasons as mentioned above.

Reply:
makes sense. thanks for your patience on this